### DISCUSSION TITLE
src refspec matches more than one

### DISCUSSION THREAD

**Discussion by AndrewSav** [jesseduffield/lazygit#3121] [Q&A] [answered]
If I have two mains on my fork, one is origin and one is upstream, I get a error message in subject when I'm trying to push. Currently I have to drop to command line from lazygit to push. What is a practical way to resolve this?

**Comment by mark2185** [comment]
Is it a new branch that you're trying to push or is it an already existing one? Does it happen on every push? What do you run in the command line to solve it?

**Reply by AndrewSav** [reply]
Fork a repo on GitHub. Follow the step on <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/configuring-a-remote-repository-for-a-fork> On you main branch create a commit and try to push, observe the error message. Drop out of git and run `git push origin main` that succeeds.

**Reply by mark2185** [reply]
Could you please post your `.gitconfig`?

Also just to confirm, the currently checked out branch _does_ have an upstream, i.e. when you press `u` on it you get something like this? The options aren't crossed out?
![image](https://github.com/jesseduffield/lazygit/assets/8365241/aa7f7c2c-09d5-42c5-b1f9-d57b02637a34)

**Reply by AndrewSav** [reply]
This is what I'm seeing pressing 'u' on the master branch. (I used word `main` in the previous post but the actual branches are called master)
![2023-11-22_09-41-57](https://github.com/jesseduffield/lazygit/assets/658865/13114966-170f-4219-946d-aff24d04a725)

`.gitconfig`:

```
[credential]
	helper = manager
[filter "lfs"]
	clean = git-lfs clean -- %f
	smudge = git-lfs smudge -- %f
	process = git-lfs filter-process
	required = true
[user]
	name = Andrew Savinykh
	email = 658865+AndrewSav@users.noreply.github.com
[core]
	autocrlf = false
[pull]
	rebase = true
[init]
	defaultBranch = main
```

Error message on push:

![2023-11-22_09-46-23](https://github.com/jesseduffield/lazygit/assets/658865/0554990e-ef9e-4d9b-97d0-5c584f509258)

.git\config:

```
[core]
	repositoryformatversion = 0
	filemode = false
	bare = false
	logallrefupdates = true
	symlinks = false
	ignorecase = true
[remote "origin"]
	url = git@github.com:AndrewSav/csvfix.git
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
[remote "upstream"]
	url = https://github.com/wlbr/csvfix.git
	fetch = +refs/heads/*:refs/remotes/upstream/*
```

**Reply by AndrewSav** [reply]
Umh: <https://github.com/wlbr/csvfix/tags>

![2023-11-22_09-57-15](https://github.com/jesseduffield/lazygit/assets/658865/a3c691e9-b9a2-4828-9d00-78bd865709f3)

Is there anything *I* can do?
Reference: <https://stackoverflow.com/questions/9378760/git-push-local-branch-with-same-name-as-remote-tag>

**Reply by mark2185** [reply, marked as answer]
I think the simplest thing would be to rename the `master` branch to `main` in your fork. 

Unset the upstream via the menu, rename the branch to `main` and push it to `origin/main`. Tried it now and it works, and requires no intervention from our side.

The tag named `refs/heads/master` is not something I have come across, so I take it this workaround will suffice even for others who find themselves in such a situation.

**Reply by AndrewSav** [reply]
@mark2185 
![image](https://github.com/jesseduffield/lazygit/assets/658865/683aae66-e8b1-4eb1-951d-d4afc7ded4b7)

Is this expected?

**Reply by mark2185** [reply]
Could you describe the steps you took?

I did something like this:
- unset upstream of branch `master` (press `u` and select `unset upstream`)
- rename with `R` to `main`
- press `P` to push
- select `upstream main` since `upstream` points to my fork

**Reply by AndrewSav** [reply]
Yes, it looks I messed something up, never mind. It's working now, thank you very much for your help with this and other queries, really appreciate it, you rock!

I pressed `u` and set upstream to the `upstream master` which I was not supposed to do, which resulted in this section:

```
[branch "main"]
	remote = upstream
	merge = refs/heads/master
```

I changed that to:

```
[branch "main"]
	remote = origin
	merge = refs/heads/main
```

And it is working so far. The upstream's function is to be able to merge back the changes in the upstream, I have not tested it but it looks like it should work too.

> select `upstream main` since `upstream` points to my fork

not sure what this bit means, according to the github page I linked for setting up fork with an upstream `upstream` is supposed to point to where you forked it from.

**Reply by mark2185** [reply]
> not sure what this bit means, according to the github page I for setting up fork with an upstream upstream is supposed to point to where you forked it from.

I didn't read it _that carefully_, my bad. :sweat_smile: 

All that matters is that you got it working! :)

P.S. update `lazygit` from time to time, latest version is `v0.40.2`

**Reply by AndrewSav** [reply]
I'm already on the latest, and update it regularly, thanks!

**Reply by mark2185** [reply]
Ah, once again, my bad, the `Upstream options` that had me confused haven't been released yet.

Sorry for the noise!
